src/utils/bundleFeedUtils.ts

```diff
@@ -0,0 +1,89 @@
+            };
+        };
+    }
+
+    export interface ITemplatesRelease {
+        functions: string;
+        bindings: string;
+        resources: string;
+    }
+
+    export async function getLatestTemplateVersion(bundleMetadata: IBundleMetadata | undefined): Promise<string> {
+        // tslint:disable-next-line: strict-boolean-expressions
+        bundleMetadata = bundleMetadata || {};
+
+        const feed: IBundleFeed = await getBundleFeed(bundleMetadata);
+        const validVersions: string[] = Object.keys(feed.bundleVersions).filter((v: string) => !!semver.valid(v));
+        // For now just get the latest release and ignore user's bundle range from host.json
+        // https://github.com/microsoft/vscode-azurefunctions/issues/1203
+        const bundleVersion: string | null = semver.maxSatisfying(validVersions, '*');
+        if (!bundleVersion) {
+            throw new Error(localize('failedToFindBundleVersion', 'Failed to find bundle version satisfying range "{0}".', bundleMetadata.version));
+        } else {
+            return feed.bundleVersions[bundleVersion].templates;
+        }
+    }
+
+    export async function getRelease(bundleMetadata: IBundleMetadata | undefined, templateVersion: string): Promise<ITemplatesRelease> {
+        const feed: IBundleFeed = await getBundleFeed(bundleMetadata);
+        return feed.templates.v1[templateVersion];
+    }
+
+    export function isBundleTemplate(template: IFunctionTemplate | IBindingTemplate): boolean {
+        return !template.isHttpTrigger && !template.isTimerTrigger;
+    }
+
+    // We access the feed pretty frequently, so cache it and periodically refresh it
+    let cachedBundleFeed: IBundleFeed | undefined;
+    let cachedUrl: string | undefined;
+    let nextRefreshTime: number = Date.now();
+
+    async function getBundleFeed(bundleMetadata: IBundleMetadata | undefined): Promise<IBundleFeed> {
+        const bundleId: string = bundleMetadata && bundleMetadata.id || defaultBundleId;
+
+        // Only use an aka.ms link for the most common case, otherwise we will dynamically construct the url
+        let url: string;
+        if (bundleId === defaultBundleId && ext.templateProvider.templateSource !== TemplateSource.Staging) {
+            url = 'https://aka.ms/AA66i2x';
+        } else {
+            const suffix: string = ext.templateProvider.templateSource === TemplateSource.Staging ? 'staging' : '';
+            url = `https://functionscdn${suffix}.azureedge.net/public/ExtensionBundles/${bundleId}/index-v2.json`;
+        }
+
+        if (!cachedBundleFeed || cachedUrl !== url || Date.now() > nextRefreshTime) {
+            const request: requestUtils.Request = await requestUtils.getDefaultRequest(url);
+            const response: string = await requestUtils.sendRequest(request);
+            cachedBundleFeed = <IBundleFeed>JSON.parse(response);
+            cachedUrl = url;
+            nextRefreshTime = Date.now() + 10 * 60 * 1000;
```

Comment:
Why limit users from checking for new templates to every 10 minutes?

Reply:
Uh so this is for the bundle feed, not the templates (related, but different). The bundle feed is not supposed to change very often (on the scale of days to weeks), but we make a lot of calls to it, so we're essentially just caching it for 10 minutes to improve performance